include/libtorrent/packet_pool.hpp

```diff
@@ -78,78 +78,82 @@ namespace libtorrent
 		std::uint8_t buf[1];
 	};
 
+
+	struct packet_deleter
+	{
+		void operator()(packet* handle) const
+		{
+			std::free(handle);
+		}
+	};
+
+	using packet_ptr = std::unique_ptr<packet, packet_deleter>;
+
+	template<std::size_t ALLOCATE_SIZE>
+	struct TORRENT_EXTRA_EXPORT packet_slab : private aux::vector<packet_ptr, std::size_t>
+	{
+		static const std::size_t allocate_size{ ALLOCATE_SIZE };
+
+		packet_slab(std::size_t limit) : m_limit(limit) { reserve(m_limit); }
+
+		void try_free(packet_ptr &p)
```

Comment:
I don't think there's a need for this to *try* to free anymore. It could just be ``free()`` and it could take the ``packet_ptr`` by value, and let it free itself in the case where it's not added back into the cache